File: ArenaApp/src/main/java/msa/arena/MainActivity.java
package msa.arena;

import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.v7.widget.AppCompatButton;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;

import com.jakewharton.rxbinding2.view.RxView;

import java.util.List;

import butterknife.BindView;
import butterknife.BindViews;
import butterknife.ButterKnife;
import msa.arena.base.BaseActivity;

public class MainActivity extends BaseActivity {

    @BindView(R.id.toolbar)
    Toolbar toolbar;

    @BindViews({R.id.btn_search_menu, R.id.btn_search_spinner, R.id.btn_movie_list})
    List<AppCompatButton> appCompatButtonList;

    @BindView(R.id.fab)
    FloatingActionButton fab;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);
        setSupportActionBar(toolbar);
    }

    @Override
    protected void bind() {

        compositeDisposable.add(RxView.clicks(fab).subscribe(o -> Snackbar.make(fab, "Thank you for visiting Arena", Snackbar.LENGTH_LONG).setAction("Action", null).show()));

        compositeDisposable.add(RxView.clicks(appCompatButtonList.get(0)).subscribe(o -> Navigator.navigateToSearchMenuActivity(this)));

        compositeDisposable.add(RxView.clicks(appCompatButtonList.get(1)).subscribe(o -> Navigator.navigateToSearchSpinnerActivity(this)));

        compositeDisposable.add(RxView.clicks(appCompatButtonList.get(2)).subscribe(o -> Navigator.navigateToMovieListActivity(this)));

    }

    @Override
    protected void unBind() {

    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: koinarc/src/main/java/com/abhi/koinarc/MainActivity.kt
package com.abhi.koinarc

import android.content.Intent
import android.os.Bundle
import android.support.design.widget.Snackbar
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import kotlinx.android.synthetic.main.activity_main.*
import org.koin.android.architecture.ext.viewModel
import org.koin.android.ext.android.inject

class MainActivity : AppCompatActivity() {

    // Inject SimplePresenter
    val presenter: SimplePresenter by inject()

    // Inject SimpleViewModel
    val viewModel: SimpleViewModel by viewModel()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbar)

        fab.setOnClickListener { view ->
            Snackbar.make(view, "Replace with your own action ${viewModel.count}", Snackbar.LENGTH_LONG)
                    .setAction("Action", null).show()
            viewModel.count += 1
            startActivity(Intent(this, TabActivity::class.java))
        }




        Log.i(this.localClassName, "Hello from Presenter = ${presenter.sayHello()}")
        Log.i(this.localClassName, "Hello from ViewModel = ${viewModel.sayHello()} with count = ${viewModel.count}")
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        return when (item.itemId) {
            R.id.action_settings -> true
            else -> super.onOptionsItemSelected(item)
        }
    }
}
